include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -134,43 +134,52 @@ __brick_walk1(_DifferenceType __n, _Function __f, ::std::true_type) noexcept
     oneapi::dpl::__internal::__brick_walk1(__n, __f, ::std::false_type{});
 }
 
-template <class _ExecutionPolicy, class _ForwardIterator, class _Function, class _IsVector>
-oneapi::dpl::__internal::__enable_if_host_execution_policy<_ExecutionPolicy, void>
-__pattern_walk1(_ExecutionPolicy&&, _ForwardIterator __first, _ForwardIterator __last, _Function __f,
-                _IsVector __is_vector,
-                /*parallel=*/::std::false_type) noexcept
+template <class _Tag, class _ExecutionPolicy, class _ForwardIterator, class _Function>
+void
+__pattern_walk1(_Tag, _ExecutionPolicy&&, _ForwardIterator __first,
```

Comment:
In result we agreed that we have a serial execution by default, by "general" tag, in case of no other overloads found, right? So, do we still need `__serial_tag`?

Reply:
If I correctly understand your question we still need to have the particular type to pass even if there is no dedicated overload for that. `select_backend` should return something and then we need to read `is_vector` from that "something":)

Reply:
Ok, I see.
In that case I would suggest renaming  `__serial_tag` to `__no_parallel_tag`, just to avoid some confusing, because an execution with  `__serial_tag::is_vector == true` is not **serial**.

Reply:
During the discussion we decided to leave it as is for now to show transition from `_allow_parallel` to two different tags.